collectors/python.d.plugin/hpssa/hpssa.chart.py

```diff
@@ -0,0 +1,402 @@
+                        adapter['cache']['present'] = value == 'True'
+                    elif key == 'Cache Status':
+                        adapter['cache']['status'] = value == 'OK'
+                    elif key == 'Cache Module Temperature (C)':
+                        adapter['cache']['temperature'] = int(value)
+                    elif key == 'Battery/Capacitor Count':
+                        adapter['battery']['count'] = int(value)
+                    elif key == 'Battery/Capacitor Status':
+                        adapter['battery']['status'] = value == 'OK'
+                else:
+                    raise HPSSAException('Cannot parse line: {}'.format(line))
+
+        return adapter
+
+    def _parse_array(self, adapter):
+        for line in self:
+            if _adapter_regex.match(line) or _array_regex.match(line) or _ignored_sections_regex.match(line):
+                self._rewind()
+                break
+
+            match = _drive_regex.match(line)
+            if match:
+                data = match.groupdict()
+                if data['logical_drive_id']:
+                    self._parse_logical_drive(adapter, int(data['logical_drive_id']))
+                else:
+                    self._parse_physical_drive(adapter, data['fqn'])
+            elif not _key_value_regex.match(line):
+                self._rewind()
+                break
+
+    def _parse_unassigned_physical_drives(self, adapter):
+        for line in self:
+            match = _drive_regex.match(line)
+            if match:
+                self._parse_physical_drive(adapter, match.group('fqn'))
+            else:
+                self._rewind()
+                break
+
+    def _parse_logical_drive(self, adapter, logical_drive_id):
+        ld = {
+            'id': logical_drive_id,
+            'status': None,
+            'status_complete': None,
+        }
+
+        for line in self:
+            if _mirror_group_regex.match(line):
+                self._parse_ignored_section()
+            elif _adapter_regex.match(line) \
+                    or _array_regex.match(line) \
+                    or _drive_regex.match(line) \
+                    or _ignored_sections_regex.match(line):
+                self._rewind()
+                break
+            else:
+                match = _key_value_regex.match(line)
+                if match:
+                    key, value = match.group('key', 'value')
+                    if key == 'Status':
+                        status_match = _status_complete_regex.match(value)
+                        if status_match:
+                            ld['status'] = status_match.group('status') == 'OK'
+                            ld['status_complete'] = float(status_match.group('percentage')) / 100
+                        else:
+                            ld['status'] = value == 'OK'
+                else:
+                    self._rewind()
+                    break
```

Comment:
`parse_logical_drive` method looks complicated to me, a lot of nested if/else. It would be nice if you could make it simpler